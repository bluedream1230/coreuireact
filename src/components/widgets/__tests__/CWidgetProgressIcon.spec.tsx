import * as React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { CWidgetProgressIcon } from '../../../index'

test('loads and displays CWidgetProgressIcon component', async () => {
  const { container } = render(<CWidgetProgressIcon/>)
  expect(container).toMatchSnapshot()
})

test('CWidgetProgressIcon customize', async () => {
    const { container } = render(
        <CWidgetProgressIcon
            className="bazinga"
            color="info"
            icon="icon"
            progressColor="warning"
            progressValue={75}
            progressWhite={true}
            title="title"
            textColor="white"
            value="value"
        >
            Test
        </CWidgetProgressIcon>
    )
    expect(container).toMatchSnapshot()
    expect(container.firstChild).toHaveClass('bazinga')
    expect(container.firstChild).toHaveClass('card')
    if(container.firstChild === null || container.firstChild.firstChild === null){
        expect(true).toBe(false)
    }else{
        expect(container.firstChild.firstChild).toHaveClass('card-body')
        expect(container.firstChild.firstChild.firstChild).toHaveClass('text-medium-emphasis-inverse')
        expect(container.firstChild.firstChild.firstChild).toHaveClass('text-end')
        expect(container.firstChild.firstChild.firstChild).toHaveClass('mb-4')
        expect(container.firstChild.firstChild.firstChild).toHaveTextContent('icon')
        expect(container.firstChild.firstChild.lastChild).toHaveClass('mt-3')
        expect(container.firstChild.firstChild.lastChild).toHaveClass('mb-0')
    }
})