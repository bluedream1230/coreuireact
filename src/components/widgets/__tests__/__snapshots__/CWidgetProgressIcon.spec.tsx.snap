// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CWidgetProgressIcon customize 1`] = `
<div>
  <div
    class="card bg-info text-white bazinga"
  >
    <div
      class="card-body"
    >
      <div
        class="text-medium-emphasis-inverse text-end mb-4"
      >
        icon
      </div>
      <div
        class="text-high-emphasis-inverse fs-4 fw-semibold"
      >
        value
      </div>
      <div
        class="text-medium-emphasis-inverse text-uppercase fw-semibold small"
      >
        title
      </div>
      <div
        class="progress progress-white mt-3 mb-0"
        style="height: 4px;"
      >
        <div
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="75"
          class="progress-bar bg-warning"
          role="progressbar"
          style="width: 75%;"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`loads and displays CWidgetProgressIcon component 1`] = `
<div>
  <div
    class="card"
  >
    <div
      class="card-body"
    >
      <div
        class="progress mt-3 mb-0"
        style="height: 4px;"
      />
    </div>
  </div>
</div>
`;
